src/compile/common.ts

```diff
@@ -77,7 +77,34 @@ export function getMarkConfig<P extends keyof MarkConfig>(prop: P, mark: Mark, c
   return config.mark[prop];
 }
 
-export function formatSignalRef(fieldDef: FieldDef<string>, specifiedFormat: string, expr: 'datum' | 'parent', config: Config, useBinRange?: boolean) {
+export function formatSignalRef(fieldDef: FieldDef<string>, specifiedFormat: string, expr: 'datum' | 'parent', config: Config, formatType: 'number' | 'time' | 'utc', channel: Channel, useBinRange?: boolean) {
+  if (channel === TEXT || channel === ROW || channel === COLUMN || channel === TOOLTIP) {
```

Comment:
Can you explain why we need to add so many channels check up here? 

The logic of this chunk is very redundant with the original group of code. 

I guess the difference is whether a channel has a scale or not, but copying the same chunk of code and run it twice is definitely no a good practice better only create conditions for necessary lines.